ext/strscan/strscan.c

```diff
@@ -620,10 +623,16 @@ strscan_do_scan(VALUE self, VALUE pattern, int succptr, int getstr, int headonly
 
     if (succptr) {
         succ(p);
+        if (succptr == 2) p->curr = p->regs.beg[0];
     }
     {
         const long length = last_match_length(p);
         if (getstr) {
+            if (getstr == 2) {
+                return extract_range(p,
+                                     0,
+                                     adjust_register_position(p, p->regs.beg[0]));
```

Comment:
This is not correct, but I'm not familiar enough with the code to know how to change this. Essentially, we just want to return "up to" (but not including) the matched regex.